source/nodejs/adaptivecards/src/card-elements.ts

```diff
@@ -2452,7 +2460,255 @@ export class MediaSource extends SerializableObject {
     }
 }
 
+export abstract class MediaPlayer {
+    private _posterUrl?: string;
+
+    abstract canPlay(): boolean;
+    abstract render(): HTMLElement;
+    abstract fetchVideoDetails(): Promise<void>;
+
+    play() {
+        // Do nothing in base implementation
+    }
+
+    get posterUrl(): string | undefined {
+        return this._posterUrl;
+    }
+
+    protected set posterUrl(value: string | undefined) {
+        this._posterUrl = value;
+    }
+
+    get selectedMediaType(): string | undefined {
+        return undefined;
+    }
+}
+
+export class HTML5MediaPlayer extends MediaPlayer {
+    private _selectedMediaType?: string;
+    private _selectedSources: MediaSource[] = [];
+    private _mediaElement?: HTMLMediaElement;
+
+    private processSources() {
+        this._selectedSources = [];
+        this._selectedMediaType = undefined;
+
+        for (const source of this.owner.sources) {
+            const mimeComponents = source.mimeType ? source.mimeType.split("/") : [];
+
+            if (mimeComponents.length === 2) {
+                if (!this._selectedMediaType) {
+                    const index = HTML5MediaPlayer.supportedMediaTypes.indexOf(mimeComponents[0]);
+
+                    if (index >= 0) {
+                        this._selectedMediaType = HTML5MediaPlayer.supportedMediaTypes[index];
+                    }
```

Comment:
The code is more readable if you add `else {continue;}`
it lets the dev knows that: hi, mediatype is not supported and not need to do selectedSource.push

Reply:
`continue` forces the code to start the next iteration in the nearest loop statement. If I added it, the next `if (mimeComponents[0] === this._selectedMediaType` statement would be skipped, which isn't what needs to happen.